### PR TITLE
    EES-3268 - add health checks to admin, content & data APIs

### PR DESCRIPTION
**Pull request by luke-h1**
This PR adds health checks to the admin, data & content API. This has been added in order to provide an endpoint that Azure can use during deployments (in an attempt to fix EES-949)

### CODE REVIEW

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Startup.cs [hunk 1]
```diff
@@ -462,11 +465,13 @@ public void ConfigureServices(IServiceCollection services)
                     }
                 });
             });
-        }
+       }
 
         // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
         public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
         {
+
```
Comment (duncan-at-hiveit) [inline src/GovUk.Education.ExploreEducationStatistics.Admin/Startup.cs:473]: Just check over the formatting in this file?

File: src/GovUk.Education.ExploreEducationStatistics.Content.Api/Startup.cs [hunk 2]
```diff
@@ -199,6 +201,8 @@ public void ConfigureServices(IServiceCollection services)
                 .AllowAnyHeader());
 
             app.UseMvc();
+            app.UseHealthChecks("/api/health");
+
```
Comment (duncan-at-hiveit) [inline src/GovUk.Education.ExploreEducationStatistics.Content.Api/Startup.cs:205]: Empty line here

File: src/GovUk.Education.ExploreEducationStatistics.Data.Api/Startup.cs [hunk 3]
```diff
@@ -236,6 +238,8 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
                 .AllowAnyHeader());
 
             app.UseMvc();
+            app.UseHealthChecks("/api/health");
+
```
Comment (duncan-at-hiveit) [inline src/GovUk.Education.ExploreEducationStatistics.Data.Api/Startup.cs:242]: Empty line here

File: src/GovUk.Education.ExploreEducationStatistics.Admin/web.config [hunk 4]
```diff
@@ -15,10 +15,12 @@
...
       </requestFiltering>
     </security>
-   <applicationInitialization
+
+    <applicationInitialization
       doAppInitAfterRestart="true"
     >
-      <add initializationPage="/"/> 
+      <add initializationPage="/api/health" 
+      />
```
Comment (duncan-at-hiveit) [inline src/GovUk.Education.ExploreEducationStatistics.Admin/web.config:23]: Can remove the newline here

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Startup.cs (hunk 1 above)
Comment (duncan-at-hiveit) [reply src/GovUk.Education.ExploreEducationStatistics.Admin/Startup.cs:473]: got some comments that are out of alignment and some empty lines

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Startup.cs [hunk 5]
```diff
@@ -103,6 +103,9 @@ public Startup(IConfiguration configuration, IHostEnvironment hostEnvironment)
         // This method gets called by the runtime. Use this method to add services to the container.
         public void ConfigureServices(IServiceCollection services)
         {
+
+            services.AddHealthChecks();
```
Comment (duncan-at-hiveit) [inline src/GovUk.Education.ExploreEducationStatistics.Admin/Startup.cs:107]: This is cool!  So do we need to implement something ourselves at /api/health or is there a pre-built one that gives back basic information on this route?  I guess at the very least it's going to error until the API is up and running.

Comment (luke-h1) [reply src/GovUk.Education.ExploreEducationStatistics.Admin/Startup.cs:107]: Nope, by default, I believe it returns a 200 OK if it's deemed to be in the `healthy state` or a `503 service unavailable` if it's deemed to be in an `unhealthy` state. This is quite a basic implementation but you could take it further by adding more granular health checks for individual services i.e. 

```
using Microsoft.Extensions.Diagnostics.HealthChecks;
using System.Data.SqlClient;
 
namespace HealthCheck.Demo
{
    public static class DbHealthCheckProvider
    {
        public static HealthCheckResult Check(string connectionString)
        {
            try
            {
                using var connection = new SqlConnection(connectionString);
                connection.Open();
                return HealthCheckResult.Healthy();
            }
            catch
            {
                return HealthCheckResult.Unhealthy();
            }
        }
    }
}
```

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Startup.cs (hunk 1 above)
Comment (luke-h1) [reply src/GovUk.Education.ExploreEducationStatistics.Admin/Startup.cs:473]: Ta, I think some formatting went awry when I used Rider

Comment (benoutram) [reply src/GovUk.Education.ExploreEducationStatistics.Admin/Startup.cs]: I've just checked this and I can still see problems with the formatting in the diff shown in Github. When I reformat with Rider there are differences.

Maybe try reformatting with Rider again?

**Review by benoutram — CHANGES_REQUESTED** [review]
Check formatting of Admin `Statup.cs` again.

**Review by benoutram — APPROVED** [review]

**Pull request merged**